### PR TITLE
Bullet v2

### PR DESCRIPTION
**Pull request by jvarley**
@mateiciocarlie 

Integrated Bullet and Graspit to a usable stable point.. There are still a few things to be done, but overall I think it is at a point where it should be pulled in, especially since it does not change any default graspit behavior.  

Working:
1)Tested with both versions of the Barrett Hand and the pr2 gripper. 
2) By default it still compiles and runs with graspit_physics.  There is a flag in graspit.pro to flip between bullet_phyiscs and graspit_physics. 

Todo Still:
1) I am not confident that break away joints are working in a way that is representative of reality atm.
2) Mico Gripper is not behaving properly, This has to do with the Mico.xml though, not the use of bullet
3) Fix BarretGlassDyn world.  The arm and hand fall apart with Bullet, but the plannerMug example world works fine.

### CODE REVIEW

File: src/world.cpp [hunk 1]
```diff
@@ -2146,23 +2158,28 @@ joint constraints, for the next time step.
...
+
+//	for (int i=0; i<numRobots; i++) {
+//		robotVec[i]->DOFController(actualTimeStep);
+//		robotVec[i]->applyJointPassiveInternalWrenches();
+//	}
+
+//	if (computeNewVelocities(actualTimeStep)) {
+//		Q_EMIT dynamicsError("LCP could not be solved.");
+//		return;
+//	}
```
Comment (mateiciocarlie) [inline src/world.cpp]: Let's get rid of all this commented out stuff, for cleanliness.

File: src/dof.cpp [hunk 2]
```diff
@@ -96,7 +96,7 @@ void DOF::updateMinMax()
   DBGP("Joint 0 min "  << minq << " max " << maxq);
   if (maxq < minq) std::swap(maxq, minq);  
   DBGP("maxq " << maxq << " minq " << minq);
-  std::list<Joint *>::iterator j;
+  std::vector<Joint *>::iterator j;
```
Comment (mateiciocarlie) [inline src/dof.cpp]: This change from list to vector has required changes in a bunch of places; what was the initial reason for it?

File: graspit-core.pro [hunk 3]
```diff
@@ -148,6 +149,10 @@ SOURCES	+= src/arch.cpp \
 	src/worldElementFactory.cpp \
 	src/math/matrix.cpp \
 	src/Collision/collisionInterface.cpp \
+        src/dynamics/dynamics.cpp \
+        src/dynamics/bulletDynamics.cpp \
```
Comment (mateiciocarlie) [inline graspit-core.pro]: I think the bulletDynamics files (both .h and .cpp) should be included here only if they are actually used - same for the graspitDynamics files. They should be guarded by bullet_dyamics{...} or graspit_dynamics{...} respectively.

File: graspit.pro [hunk 4]
```diff
@@ -13,11 +13,15 @@ LANGUAGE	= C++
 #comment out this line for compiling in Release mode
 #usually, compiling in Release mode delivers a significant gain in performance
 #in MS Visual Studio *also* set the project mode to Release
-#CONFIG += graspitdbg
+CONFIG += graspitdbg
```
Comment (mateiciocarlie) [inline graspit.pro]: let's not let the switch to debug sneak in by default.

**Comment by mateiciocarlie** [comment]
Looks great, almost ready to go. Just a few small comments... I guess it will also need merging form teh other PRs we just merged...

File: src/dof.cpp (hunk 2 above)
Comment (jvarley) [inline src/dof.cpp]: I kept getting some weird runtime errors while using the list iterator that I couldn't figure out for the life of me.  I can try to switch it back to a list.

File: src/world.cpp (hunk 1 above)
Comment (jvarley) [inline src/world.cpp]: done.

File: graspit-core.pro (hunk 3 above)
Comment (jvarley) [inline graspit-core.pro]: done.

File: graspit.pro (hunk 4 above)
Comment (jvarley) [inline graspit.pro]: done.

**Comment by jvarley** [comment]
Made all the small changes you mentioned, kept with std::vector for Joint list rather than std::list, as I can't figure out why it segfaults with std list.

**Pull request merged**